version: '3.2'

volumes:
  postgres_data_local: {}
  postgres_backup_local: {}
  django_data: {}


services:
  django:
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    image: phame_api01_local_django
    depends_on:
      - postgres
    volumes:
      - .:/app
#      - django_data:/app/phame_api01/media
    env_file:
      - ./.envs/.local/.django
      - ./.envs/.local/.postgres
    ports:
      - "8000:8000"
    command: /start

  postgres:
    build:
      context: .
      dockerfile: ./compose/production/postgres/Dockerfile
    image: phame_api01_production_postgres
    volumes:
      - postgres_data_local:/var/lib/postgresql/data
      - postgres_backup_local:/backups
    env_file:
      - ./.envs/.local/.postgres
#  nginx:
#    image: nginx
#    volumes:
#     - ./compose/local/nginx/conf.d:/etc/nginx/conf.d
#    ports:
#     - "8080:80"


  phame:
    build:
      context: .
      dockerfile: ./compose/local/PhaME-1/Dockerfile
    image: phame_api01_phame-1
    hostname: phame_host
    ports:
      - 5000:5000
    volumes:
      - .:/phame_api01
#      - django_data:/app/phame_api01/media
#    command: source /PhaME-1/thirdParty/miniconda/bin/activate phame
#    entrypoint: [ "/bin/bash", "-c", "source activate phame"]
#    stdin_open: true
#    tty: true
